### ISSUE TITLE
Support for Serverless Application Model Cloudformation Templates

### ISSUE THREAD

**Issue by bundyfx** [awslabs/cfn-python-lint#15]
Running the tool now will give back errors relating to standard syntax use in SAM based Cloudformation templates such as:

```
E1001 Top level item Globals isn't valid
E1012 Ref ServerlessRestApi not found as a resource or parameter
E1019 String parameter ServerlessRestApi not found in string for Outputs/ApiURL/Value/Fn::Sub
```

In this case, It becomes a little tricky as resources like `ServerlessRestApi` might not be statically defined in the template but rather dynamically built to support a certain functions event properties.

**Comment by kddejong** [comment]
For Globals I wrote the rules around https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/template-anatomy.html.  I can't seem to find a lot on *Globals*... is that a Transform thing as well?

Transform logic hasn't been done yet.  Our starting focus was around the CloudFormation Resource Specifications which doesn't include Transforms.  That being said I have a few ideas on how we could handle this.  I'm not sure we will be able to tackle all the rules with a serverless transform right away but I'm hoping we could handle Fn::GetAtt and Ref.  It would be nice if there was a Serverless CloudFormation Spec existed.

Going to mark this for enhancement.
**In Scope:** Support Ref, GetAtt for Serverless Transforms

**Out of Scope:**  
- Include Transform
- Serverless resource properties.  Ideally there would be an AWS generated spec or we may have to look into creating our own for cfn-lint.

**Comment by bundyfx** [comment]
Awesome - keep up the great work!

**Comment by kddejong** [comment]
Resolved in version 0.0.10.
